daemon/algod/api/server/v2/test/handlers_test.go

```diff
@@ -208,6 +208,55 @@ func addBlockHelper(t *testing.T) (v2.Handlers, echo.Context, *httptest.Response
 	return handler, c, rec, stx, releasefunc
 }
 
+func getBlockHashTest(t *testing.T, blockNum uint64, expectedCode int) {
+}
+
+func TestGetBlockHash(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	t.Parallel()
+
+	handler, c, rec, _, _, releasefunc := setupTestForMethodGet(t)
+	defer releasefunc()
+
+	err := handler.GetBlockHash(c, 0)
+	require.NoError(t, err)
+	require.Equal(t, 200, rec.Code)
+
+	c, rec = newReq(t)
+	err = handler.GetBlockHash(c, 1)
+	require.NoError(t, err)
+	require.Equal(t, 404, rec.Code)
+}
+
+func TestGetBlockGetBlockHash(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	t.Parallel()
+
+	handler, c, rec, _, releasefunc := addBlockHelper(t)
+	defer releasefunc()
+
+	// Validate that the block returned from GetBlock has the same hash that is returned via GetBlockHash
+	format := "json"
+	err := handler.GetBlock(c, 1, generatedV2.GetBlockParams{Format: &format})
+	require.NoError(t, err)
+	require.Equal(t, 200, rec.Code)
+	blkResponse := struct {
+		Block bookkeeping.Block `codec:"block"`
+	}{}
+	// err = json.Unmarshal(rec.Body.Bytes(), &blkResponse)
+	err = protocol.DecodeJSON(rec.Body.Bytes(), &blkResponse)
+	require.NoError(t, err)
+
+	c, rec = newReq(t)
+	err = handler.GetBlockHash(c, 1)
+	require.NoError(t, err)
+	require.Equal(t, 200, rec.Code)
+	var blkHashResponse generatedV2.BlockHashResponse
+	err = protocol.DecodeJSON(rec.Body.Bytes(), &blkHashResponse)
+	require.NoError(t, err)
+	require.Equal(t, crypto.HashObj(blkResponse.Block.BlockHeader).String(), blkHashResponse.BlockHash)
+}
```

Comment:
Maybe assert that the hash doesn't contain AAAAAAAA (the zero hash)

When I did this, I grabbed two blocks and asserted that block 2's "prev" was the equal to block 1's hash
https://github.com/algorand/go-algorand/compare/master...cce:go-algorand:add-v2-block-hash#diff-c9875030556c016ef35a4b47373f5005ee03b961a82157bed9ef734881edddc3R253-R268

Reply:
I'm pretty conflicted about the way "prev" is implemented...Because it's [wrapped in a `BlockHash`](https://github.com/algorand/go-algorand/blob/master/data/bookkeeping/block.go#L249) object which overrides [the `String()` method](https://github.com/algorand/go-algorand/blob/master/data/bookkeeping/prettyprinting.go#L28) by prepending `"blk-"`, these would produce different outputs
```
BlockHash(crypto.Digest([]byte{"AAAAAA"})).String()
crypto.Digest([]byte{"AAAAAA"}).String()
```

On the other hand, the [crypto.Digest String() method](https://github.com/algorand/go-algorand/blob/master/crypto/util.go#L56) does a base-32 encoding of the hash bytes when converting to a string.

I'm tempted to just return bytes from `GetBlockHash` and let the user do whatever conversion they want, but on the other hand I agree that if a returned BlockHash didn't match up with what was in "prev" it would be very confusing for the end user.

Reply:
I'm going to be out for a while so I'll leave the decision up to @winder but in my opinion, returning `"blk-1235"` as a hash value from this API feels strange.

Reply:
Oh good point, which format is "correct" for this purpose, the `blk-XXX` hash version or just `XXX`? It seems the REST API is already using the `blk-` prefixed version for the existing endpoint, so there's one point for consistency. I don't really know much about where else these `blk-` prefix hashes appear

Reply:
My only contributions here are some observations:
1. algoexplorer seems to remove the `blk-`.
2. Indexer b64 encodes it.

From this, it seems like everyone agrees that `blk-` is a little strange so this is a consistency question. For consistency, I think we should select the B32 encoding over B64. If someone wants to compare them they'll have to remove the `blk-` prefix.